hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@dtarb @horsburgh @selimnairb One way to tackle this is to retrieve all these resources with failed crossref deposition in the same DOI activation monitoring task already scheduled to run once every midnight to retry deposition again. This will minimize performance impact resulting from too frequent retries in case crossref system is not responsive for some time although the possibility for such event is small. On resource front page, we show a message on the top indicating the resource has been requested for publication, but there is a system issue with crossref, and we are retrying the request until it succeeds and will notify users when status changes. We will not change resource sharing status and metadata until metadata deposition with crossref succeeds. What do you think? If you think this is what we should do, please let me know the message text to be displayed on the landing page of these resources with failed metadata deposition with crossref.